blocks/api/parser.js

```diff
@@ -123,13 +123,54 @@ export function isValidBlock( rawContent, blockType, attributes ) {
 
 	const isValid = ( actual === expected );
 
-	if ( ! isValid && 'development' === process.env.NODE_ENV ) {
-		// eslint-disable-next-line no-console
-		console.error(
-			'Invalid block parse\n' +
-				'\tExpected: ' + expected + '\n' +
-				'\tActual:   ' + actual
-		);
+	if ( ! isValid ) {
+		// NOTE: DO NOT COLLAPSE THESE CONDITIONS OR CONVERT COMMONJS TO ES2015
+		// IMPORTS. They are intentionally written as such because minification
+		// will drop this unreachable block in production environments.
+		if ( 'production' !== process.env.NODE_ENV ) {
+			const chalk = require( 'chalk' );
+			let message = 'Invalid block parse for ';
+
+			// Chalk automatically detects its environment to determine whether
+			// ANSI codes are supported. In these environments we provide a nicely
+			// colored error message for invalid parse.
+			//
+			// TODO: Teach browsers to ANSI
+			if ( chalk.enabled ) {
+				const { diffLines } = require( 'diff' );
+				const diff = diffLines( actual, expected );
+				message += chalk.bold( blockType.name ) + ':\n\n';
+				message += chalk.bgGreen( 'Expected' ) + ' ' + chalk.bgRed( 'Actual' ) + '\n\n';
+
+				message += diff.reduce( ( result, part ) => {
+					let { value } = part;
+					if ( part.removed ) {
+						value = value.replace( /\n$/, '' );
+						value = value.replace( /^/gm, '- ' );
+						value = chalk.red( value );
+					} else if ( part.added ) {
+						value = value.replace( /\n$/, '' );
+						value = value.replace( /^/gm, '+ ' );
+						value = '\n' + value;
+						value = chalk.green( value );
+						value += '\n';
+					}
+
+					return result + value;
+				}, '' );
+			} else {
+				message += blockType.name + ':\n\n' +
+					'Expected:\n\n' + expected + '\n\n' +
+					'Actual:\n\n' + actual;
+			}
+
+			if ( 'test' === process.env.NODE_ENV ) {
+				throw new Error( message );
+			} else {
+				// eslint-disable-next-line no-console
+				console.warn( message );
+			}
+		}
```

Comment:
The `NODE_ENV` condition wasn't so much for testing the parser as much as it was to ensure the fixtures testing would actually log the failure if the parse was invalid (`console.warn` is not sufficient).

Instead, in cdde9c4 I've changed it so that the error is thrown in all cases. It is handled by a `try` / `catch` in `createBlockWithFallback` which assigns the error into the block. With a reintroduction of the invalid blocks, the fixtures testing determines whether the parse _should_ have failed and otherwise surfaces the error to Jest.

I went back and forth on this a bit, trying to decide if throwing an error was the right path. It's otherwise difficult to detect the failure in fixtures testing... except with overriding `console.error`, which felt a bit hacky. I think they're legitimate errors and should be treated as such.

The downside is that because the error is handled, it is no longer logged to the console in the browser. But because the error is now stored into the block, I think we have some better opportunities to surface the message through the UI, rather than through the console (even in the "Invalid Block" dialog).

To simplify the formatting, I've removed the `chalk.isEnabled` split and just use chalk in all cases. It degrades gracefully to `identity` when not enabled.